### PR TITLE
Extract proptest infrastructure into merge_parity_proptest.rs

### PR DESCRIPTION
**Pull request by ahrav**
## Summary by CodeRabbit

* **New Features**
  * Hardened PostgreSQL backend with deterministic content-addressed IDs, enforced schema constraints, and safer migrations.

* **Tests**
  * Added extensive property-based parity tests to validate SQL upsert/merge behavior against in-memory logic.
  * Improved test scaffolding and deterministic fixtures for boundary conditions.

* **Refactor**
  * Reorganized persistence modules, consolidated error handling, and clarified public/internal boundaries.

### CODE REVIEW

File: crates/gossip-done-ledger-postgres/src/tests/merge_parity_proptest.rs [hunk 1]
```diff
@@ -0,0 +1,771 @@
...
+                    existing_duration,
+                    later_finished_gap,
+                    incoming_started_raw,
+                ),
+            )| {
+                let existing_started_at = base_offset + existing_started_at_raw;
+                let existing_finished_at = existing_started_at + existing_duration;
+                let incoming_finished_at = existing_finished_at + later_finished_gap;
+                let incoming_started_at =
+                    (base_offset + incoming_started_raw).min(incoming_finished_at);
```
[the remark this answers is not in the document]
Comment (ahrav) [reply crates/gossip-done-ledger-postgres/src/tests/merge_parity_proptest.rs]: The bias observation is correct — when `base_offset=0` and the combined timestamps are small, `incoming_started_raw.min(incoming_finished_at)` collapses most values to `incoming_finished_at`.

However, this is not masking merge bugs: the `LaterFinishedAt` winner selection does not depend on `started_at` at all (that branch fires on `incoming.finished_at > existing.finished_at` alone). The `started_at` dimension is specifically exercised by `arb_later_started_case`, and the `General` strategy provides unconstrained coverage.

Applied the improvement anyway since it is clearly better test design — switched from `prop_map` + clamp to `prop_flat_map` drawing `incoming_started_at` uniformly from `0..=incoming_finished_at`. This eliminates the wasted input space where thousands of distinct `incoming_started_raw` values all collapse to the same clamped output.
